src/ForeignCallbacks.jl

```diff
@@ -114,8 +114,10 @@ mutable struct ForeignCallback{T}
         end
         this = new{T}(stack, cond, task)
         finalizer(this) do this
-            close(this.cond)
-            # TODO: free queue we are leaking at least one node here
+            Threads.@spawn begin
+                close(this.cond)
```

Comment:
```suggestion
                close(cond)
```
And then `cond = this.cond` outside the `@spawn`. Otherwise we leak a reference to the finalized objects in the new task.

Reply:
Done